post_office/sanitizer.py

```diff
@@ -0,0 +1,144 @@
+            'table': ['class', 'id', 'style', 'align', 'bgcolor', 'border', 'cellpadding', 'cellspacing', 'dir', 'frame', 'rules', 'width'],
+            'tbody': ['class', 'id', 'style'],
+            'tfoot': ['class', 'id', 'style'],
+            'td': ['class', 'id', 'style', 'abbr', 'align', 'bgcolor', 'colspan', 'dir', 'height', 'lang', 'rowspan', 'scope', 'style', 'valign', 'width'],
+            'th': ['class', 'id', 'style', 'abbr', 'align', 'background', 'bgcolor', 'colspan', 'dir', 'height', 'lang', 'scope', 'style', 'valign', 'width'],
+            'thead': ['class', 'id', 'style'],
+            'tr': ['class', 'id', 'style', 'align', 'bgcolor', 'dir', 'style', 'valign'],
+            'u': ['class', 'id', 'style'],
+            'ul': ['class', 'id', 'style', 'dir', 'type'],
+        },
+        strip=True,
+        strip_comments=True,
+        styles=[
+            'border', 'border-top', 'border-right', 'border-bottom', 'border-left',
+            'border-radius',
+            'box-shadow',
+            'height',
+            'margin', 'margin-top', 'margin-right', 'margin-bottom', 'margin-left',
+            'padding', 'padding-top', 'padding-right', 'padding-bottom', 'padding-left',
+            'width',
+            'max-width',
+            'min-width',
+            'border-collapse',
+            'border-spacing',
+            'caption-side',
+            'empty-cells',
+            'table-layout',
+            'direction',
+            'font',
+            'font-family',
+            'font-style',
+            'font-variant',
+            'font-size',
+            'font-weight',
+            'letter-spacing',
+            'line-height',
+            'text-align',
+            'text-decoration',
+            'text-indent',
+            'text-overflow',
+            'text-shadow',
+            'text-transform',
+            'white-space',
+            'word-spacing',
+            'word-wrap',
+            'vertical-align',
+            'color',
+            'background',
+            'background-color',
+            'background-image',
+            'background-position',
+            'background-repeat',
+            'bottom',
+            'clear',
+            'cursor',
+            'display',
+            'float',
+            'left',
+            'opacity',
+            'outline',
+            'overflow',
+            'position',
+            'resize',
+            'right',
+            'top',
+            'visibility',
+            'z-index',
+            'list-style-position',
+            'list-style-tyle',
+        ],
+    ))
+except ImportError:
+    # if bleach is not installed, handle rendered HTML as plain text
+    clean_html = lambda body: mark_safe(format_lazy('<p><em>({heading})</em></p><div>{body}</div>',
+                                                    heading=gettext_lazy("stripping all HTML tags"),
+                                                    body=escape(strip_tags(body))))
```

Comment:
In addition to this, let's display a text informing the user if they want the user to see the nice and properly rendered HTML email, they need to install `bleach`.

If you're putting the body content through `escape`, might as well remove `strip_tags()` so users would see the actualy plain text preview of the HTML email.

Reply:
> If you're putting the body content through escape
well, then we don't have to mark the text as safe at all and Django will do it for us.

I tested it with my own generated mails. If you escape everything without stripping HTML tags, you get an unreadable mess of HTML, packed into one line. Therefore stripping first and then escaping at least leaves the text readable, although not very well.

I will a not to install `bleach`.